### PR TITLE
copier: update buffer format on bind

### PR DESCRIPTION
**Pull request by iganakov**
Windows driver sends SET_SINK_FORMAT IPC to configure Copier module sink processing parameters before bind when there is no processing buffer between connected components.

Add Copier bind operation to update buffer format on BIND IPC.

### CODE REVIEW

File: src/audio/copier/copier.c [hunk 1]
```diff
@@ -990,6 +977,29 @@ static int copier_get_hw_params(struct comp_dev *dev, struct sof_ipc_stream_para
...
+
+	/* update corresponding sink format */
+	list_for_item(sink_list, &dev->bsink_list) {
+		int sink_id;
+
+		sink = container_of(sink_list, struct comp_buffer, source_list);
+		sink_id = IPC4_SRC_QUEUE_ID(buf_get_id(sink));
+		if (sink_id == bu->extension.r.src_queue) {
+			ipc4_update_buffer_format(sink, &cd->out_fmt[sink_id]);
+			break;
```
Comment (lyakh) [inline src/audio/copier/copier.c:996]: you could use `return 0;` here and return a `-ENODEV` error below, which would mean, that no suitable sink ID was found

Comment (lgirdwood) [reply src/audio/copier/copier.c:996]: If we return an error lets make sure we an error entry in the log to help debug. Thanks !

Comment (iganakov) [reply src/audio/copier/copier.c]: Sure. Thanks!

**Review by lgirdwood — APPROVED** [review]

**Review by dnikodem — APPROVED** [review]

**Review by tlissows — APPROVED** [review]

**Comment by iganakov** [comment]
I'm closing this PR. All issues are addressed in a new one https://github.com/thesofproject/sof/pull/9276

**Pull request closed without merging**